torchgeo/datasets/wms.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# Author: Ian Turton, Glasgow University ian.turton@gla.ac.uk
+
+"""A simple class to fetch WMS images."""
+from io import BytesIO
+from typing import Any, Callable, Optional, Union
+
+import torchvision.transforms as transforms
+from owslib.map.wms111 import ContentMetadata
+from owslib.wms import WebMapService
+from PIL import Image
+from rasterio.coords import BoundingBox
+from rasterio.crs import CRS
+from rasterio.errors import CRSError
+from rtree.index import Index, Property
+
+from torchgeo.datasets import GeoDataset
+
+
+class WMSDataset(GeoDataset):
+    """Allow models to fetch images from a WMS (at a good resolution)."""
```

Comment:
Also, can you add:
```rst
.. versionadded:: 0.6
```
to the docstring? This will document what version of TorchGeo is required to use this feature.